### DISCUSSION TITLE
Experience building IPOPT without Fortran?

### DISCUSSION THREAD

**Discussion by whart222** [coin-or/Ipopt#445] [Q&A] [answered]
Anyone have experience compiling IPOPT without a Fortran compiler?  The examples online that I've seen seem old and incomplete.

**Comment by tkralphs** [comment]
Hey @whart222, good to see you here. Could you clarify what you mean exactly? The Fortran compiler should only be required for building the dependencies.

**Comment by svigerske** [comment]
Ipopt requires Blas, Lapack, and a linear solver. Some of these are implemented in Fortran.

But there is almost no Fortran code in Ipopt itself. Only when you use MA28 or want the Fortran interface, there is a need for a Fortran compiler. Ipopt's configure will check for a Fortran compiler, but will continue if none is found. In that case, MA28 cannot be used and the Fortran interface is not build.

**Comment by whart222** [comment, marked as answer]
To clarify, I'm trying to do an entire build of Ipopt and its dependencies *without* a Fortran compiler.  I tried using f2c, and was able to compile HSL and BLAS files.  But LAPACK includes a variety of F95 fortran files.  :(

**Reply by svigerske** [reply]
Maybe you find some C implementation of LAPACK? Are ATLAS or OpenBLAS in C only? Or use Intel MKL.
